plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -299,17 +299,17 @@ private List<AddFileEntry> retrieveActiveFiles(
                     }
                 }
 
-                List<AddFileEntry> activeFiles = loadActiveFiles(tableSnapshot, metadataEntry, protocolEntry, TupleDomain.all(), Optional.of(alwaysTrue()), session);
-                return new DeltaLakeDataFileCacheEntry(tableSnapshot.getVersion(), activeFiles);
+                Iterator<AddFileEntry> activeFilesIterator = loadActiveFiles(tableSnapshot, metadataEntry, protocolEntry, TupleDomain.all(), Optional.of(alwaysTrue()), session);
+                return new DeltaLakeDataFileCacheEntry(tableSnapshot.getVersion(), ImmutableList.copyOf(activeFilesIterator));
```

Comment:
There is a another `ImmutableList.copyOf` inside constructor. Can you change parameter to Iterator?

Reply:
Refactored the code